plugins/MosaicCamera/src/MosaicCamera.cpp

```diff
@@ -0,0 +1,560 @@
+            cameras[name].ra = conf->value("ra").toDouble();
+            cameras[name].dec = conf->value("dec").toDouble();
+            cameras[name].rotation = conf->value("rotation").toDouble();
+        }
+    }
+    conf->endArray();
+    conf->endGroup();
+}
+
+void MosaicCamera::saveSettings() const
+{
+    conf->beginGroup("MosaicCamera");
+    conf->setValue("currentCamera", currentCamera);
+    conf->setValue("showButton", getFlagShowButton());
+    conf->setValue("enabled", flagShowMosaicCamera);
+    conf->beginWriteArray("cameraVisibility");
+    int i = 0;
+    for (auto it = cameras.constBegin(); it != cameras.constEnd(); ++it) {
+        conf->setArrayIndex(i++);
+        conf->setValue("name", it.key());
+        conf->setValue("visible", it.value().visible);
+        conf->setValue("ra", it.value().ra);
+        conf->setValue("dec", it.value().dec);
+        conf->setValue("rotation", it.value().rotation);
+    }
+    conf->endArray();
+    conf->endGroup();
+}
+
+/*************************************************************************
+ Reimplementation of the getCallOrder method
+*************************************************************************/
+double MosaicCamera::getCallOrder(StelModuleActionName actionName) const
+{
+	if (actionName==StelModule::ActionDraw)
+		return StelApp::getInstance().getModuleMgr().getModule("NebulaMgr")->getCallOrder(actionName)+10.;
+	return 0;
+}
+
+
+/*************************************************************************
+ Init our module
+*************************************************************************/
+void MosaicCamera::init()
+{
+    Q_INIT_RESOURCE(MosaicCamera);
+
+    initializeUserData();
+    loadBuiltInCameras();
+
+    addAction("actionShow_MosaicCamera", N_("Mosaic Camera"), N_("Show Mosaic Camera"), "enabled", "");
+    addAction("actionShow_MosaicCamera_dialog", N_("Mosaic Camera"), N_("Show settings dialog"), configDialog, "visible");
+    addAction("actionSetRADecToView", N_("Mosaic Camera"), N_("Set RA/Dec to current view"), "setRADecToView()", "");
+    addAction("actionIncrementRotation", N_("Mosaic Camera"), N_("Increment rotation"), "incrementRotation()", "");
+    addAction("actionDecrementRotation", N_("Mosaic Camera"), N_("Decrement rotation"), "decrementRotation()", "");
+    addAction("actionNextCamera", N_("Mosaic Camera"), N_("Next camera"), "nextCamera()", "");
+    addAction("actionPreviousCamera", N_("Mosaic Camera"), N_("Previous camera"), "previousCamera()", "");
+
+    loadSettings();
+
+    if (currentCamera == "")
+    {
+        setCurrentCamera(cameraOrder[0]);
+    }
+}
+
+void MosaicCamera::enableMosaicCamera(bool b)
+{
+    if (b!=flagShowMosaicCamera)
+    {
+        flagShowMosaicCamera = b;
+        emit flagMosaicCameraVisibilityChanged(b);
+    }
+}
+
+void MosaicCamera::initializeUserData()
+{
+    userDirectory = StelFileMgr::getUserDir()+"/modules/MosaicCamera/";
+    QDir dir(userDirectory);
+    if (!dir.exists())
```

Comment:
Maybe also test for empty dir, when user only deletes contents in an attempt to reset.